File: AED-2/Arvores/bst/TAD-flexivel/matriz-flexivel/c/soma.c
Matriz* somarMatrizes(Matriz* a, Matriz* b) {
    Matriz* soma = criarMatriz(a->linhas, a->colunas);
    Celula *pa = a->inicio, *pb = b->inicio, *ps = soma->inicio;

    for (int i = 0; i < a->linhas; i++) {
        Celula *ca = pa, *cb = pb, *cs = ps;
        for (int j = 0; j < a->colunas; j++) {
            cs->elemento = ca->elemento + cb->elemento;
            ca = ca->dir;
            cb = cb->dir;
            cs = cs->dir;
        }
        pa = pa->inf;
        pb = pb->inf;
        ps = ps->inf;
    }

    return soma;
}

File: AED-2/Arvores/bst/TAD-flexivel/matriz-flexivel/java/soma.java
package matriz.java;

public class soma {
    static Matriz somar(Matriz a, Matriz b) {
        Matriz res = new Matriz(a.linhas, a.colunas);

        Celula pa = a.inicio, pb = b.inicio, pr = res.inicio;

        for (int i = 0; i < a.linhas; i++) {
            Celula ca = pa, cb = pb, cr = pr;
            for (int j = 0; j < a.colunas; j++) {
                cr.elemento = ca.elemento + cb.elemento;
                ca = ca.dir;
                cb = cb.dir;
                cr = cr.dir;
            }
            pa = pa.inf;
            pb = pb.inf;
            pr = pr.inf;
        }

        return res;
    }

}
